pulpcore/plugin/stages/artifact_stages.py

```diff
@@ -386,7 +386,9 @@ async def _needed_remote_artifacts(self, batch):
                     key = f"{str(content_artifact.pk)}-{str(d_artifact.remote.pk)}"
                     needed_ras[key] = remote_artifact
 
-        return list(needed_ras.values())
+        nras_vals = list(needed_ras.values())
+        nras_vals.sort(key=lambda x: x.sha256)
+        return nras_vals
```

Comment:
```suggestion
        return list(needed_ras.values()).sort(key=lambda x: x.sha256)
```
I don't see the point in introducing a variable that is abbreviated this much here. (I first read "nevra"...)

Also maybe add one of those "why" comments here, too.

Reply:
I don't think this will work. I think you'd need to do:

```suggestion
        return sorted(list(needed_ras.values()), key=lambda x: x.sha256)
```

Reply:
list() returns a new list-object, and list.sort sorts in-place - so it should work.

Note that I'm about to submit a change to this PR, since pulpcore just merged a PR that I have to adapt to, in this exact place. Hang on a couple.

Reply:
@ggainey David is right, `.sort()` has no return value, so if you return `${ list stuff }.sort()` then you will be returning `None` instead of the sorted list.

```
In [5]: lst = ['a', 'd', 'c', 'b']                                                                  

In [6]: print(lst.sort())                                                                           
None

```

Reply:
`sorted` sounds good.

Reply:
@ggainey My comment was directed at the wrong person, sorry :)  Your original code splits it across multiple lines correctly.